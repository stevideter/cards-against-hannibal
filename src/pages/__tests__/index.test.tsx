jest.mock('swr');
import renderer from 'react-test-renderer';
import useSWR from 'swr';
import Home from '..';

const mockUseSWR = useSWR as jest.Mock;

describe('Home', () => {
    it('renders correctly', () => {
        mockUseSWR.mockReturnValue({
            data: ['one', 'two'],
        });

        const tree = renderer.create(<Home />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
