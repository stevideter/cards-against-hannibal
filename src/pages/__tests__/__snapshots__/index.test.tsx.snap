// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="jsx-3443756320 container"
>
  <main
    className="jsx-3443756320"
  >
    <h1
      className="jsx-3443756320 title"
    >
      Welcome to Cards Against Hannibal
    </h1>
    <p
      className="jsx-3443756320 description"
    >
      A Cards Against Humanity clone in the spirit of Pretend You're Xyzzy
    </p>
    <p
      className="jsx-3443756320"
    >
      Hello 
    </p>
    <label>
      Enter Nickname:
    </label>
    <input
      onChange={[Function]}
      type="text"
    />
    <a
      className="jsx-3443756320"
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      About
    </a>
    <div
      className="jsx-3443756320 grid"
    >
      <a
        className="jsx-3443756320 card"
        href="/game/1"
      >
        <h3
          className="jsx-3443756320"
        >
          Start a new game â†’
        </h3>
        <p
          className="jsx-3443756320"
        >
          Start a game
        </p>
      </a>
    </div>
    <div
      className="jsx-3443756320 grid"
    >
      <a
        href="/game/one"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Game 
        one
      </a>
      <a
        href="/game/two"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Game 
        two
      </a>
    </div>
  </main>
</div>
`;
